src/aiida/cmdline/commands/cmd_group.py

```diff
@@ -145,41 +145,164 @@ def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+    '-c',
+    '--contains',
+    type=click.STRING,
+    default=None,
+    help='Add a filter to delete only groups for which the label contains STRING.',
+)
+@options.NODE(help='Delete only the groups that contain a node.')
 @options.FORCE()
 @click.option(
     '--delete-nodes', is_flag=True, default=False, help='Delete all nodes in the group along with the group itself.'
 )
 @options.graph_traversal_rules(GraphTraversalRules.DELETE.value)
 @options.DRY_RUN()
 @with_dbenv()
-def group_delete(group, delete_nodes, dry_run, force, **traversal_rules):
-    """Delete a group and (optionally) the nodes it contains."""
+def group_delete(
+    groups,
+    delete_nodes,
+    dry_run,
+    force,
+    all_users,
+    user,
+    type_string,
+    past_days,
+    startswith,
+    endswith,
+    contains,
+    node,
+    **traversal_rules,
+):
+    """Delete groups and (optionally) the nodes it contains."""
     from aiida import orm
     from aiida.tools import delete_group_nodes
 
-    if not (force or dry_run):
-        click.confirm(f'Are you sure you want to delete {group}?', abort=True)
-    elif dry_run:
-        echo.echo_report(f'Would have deleted {group}.')
+    if groups and (all_users or user or type_string or past_days or startswith or endswith or contains or node):
+        echo.echo_critical('Cannot specify both GROUPS and any of the other filters.')
+        return
+
+    if not groups:
+        import datetime
+        from functools import reduce
+        from operator import concat
+
+        from tabulate import tabulate
+
+        from aiida.common import timezone
+        from aiida.common.escaping import escape_for_sql_like
+
+        builder = orm.QueryBuilder()
+        filters = {}
 
-    if delete_nodes:
+        if '%' in type_string or '_' in type_string:
+            filters['type_string'] = {'like': type_string}
+        else:
+            filters['type_string'] = type_string
 
-        def _dry_run_callback(pks):
-            if not pks or force:
-                return False
-            echo.echo_warning(f'YOU ARE ABOUT TO DELETE {len(pks)} NODES! THIS CANNOT BE UNDONE!')
-            return not click.confirm('Do you want to continue?', abort=True)
+        # Creation time
+        if past_days:
+            filters['time'] = {'>': timezone.now() - datetime.timedelta(days=past_days)}
 
-        _, nodes_deleted = delete_group_nodes([group.pk], dry_run=dry_run or _dry_run_callback, **traversal_rules)
-        if not nodes_deleted:
-            # don't delete the group if the nodes were not deleted
+        # Query for specific group labels
+        filters['or'] = []
+        if startswith:
+            filters['or'].append({'label': {'like': f'{escape_for_sql_like(startswith)}%'}})
+        if endswith:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(endswith)}'}})
+        if contains:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(contains)}%'}})
```

Comment:
I wonder if these should actually be `and` clauses? If a user specifies more than 1, I think they want all matches to match all filters, right?

Reply:
I'm not sure, I think it's ok to keep it them all the same behavior as in `verdi group list` to avoid confusion.

Reply:
ok